Discussion/Discussion/Presentation/DiscussionTopics/DiscussionTopicsView.swift

```diff
@@ -160,7 +162,16 @@ public struct DiscussionTopicsView: View {
                                                 }
                                             }
                                         }
-                                        
+                                    } else if viewModel.isShowProgress == false {
+                                        FullScreenErrorView(
+                                            type: .noCourseDates(
+                                                error: DiscussionLocalization.Error.unableToLoadCourseContent,
+                                                image: CoreAssets.information.swiftUIImage
```

Comment:
Instead of passing message and icon from everywhere, how about just setting the type and define a method in `FullScreenErrorView` for message and icon. You can keep a generic type that will take message and icon as in input for future.

Reply:
The `FullScreenErrorView` is utilized across various modules, with its implementation located in the `Core` module to enable application-wide usage. Embedding messages within the full screen would necessitate moving module-specific errors into the core module, which is not ideal. Therefore, I suggest passing the message and icon instead.